docs/migration_guide.md

```diff
@@ -0,0 +1,135 @@
+# Migration Guide
+
+## 0.0.X -> 0.1.0 (2025-02-07)
+
+### Summary
+
+- Tokenization has been removed.
+- Non-chat LLMs have been removed.
+- `ChatLLM` class was replaced by `ChatModel` class and embedding functionality has been replaced by `EmbeddingModel` class.
+- `BaseMessage` class was replaced by `Message` and its subtypes (`UserMessage`, `AssistantMessage`, `SystemMessage`, `ToolMessage`).
+- `TokenMemory` no longer uses `LLM` instance to infer `maxTokens`, user needs to do that manually (if needed).
+- The`IBMvLLM` adapter has been removed.
+
+### Models
+
+#### Before
```

Comment:
It would be nice if there was a way to do the before/after side by side, as that might let users better see what's changed. (don't think that needs to be added in this PR, as that would probably require images or formatting beyond what github markdown can do)